csv-data-management/extract_data_from_csv.py

```diff
@@ -0,0 +1,69 @@
+import requests
+import pandas as pd
+import numpy as np
+from os.path import join
+import json
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.cyops_utilities.builtins import download_file_from_cyops
+from integrations.crudhub import make_request
+from .constants import LOGGER_NAME
+
+logger = get_logger(LOGGER_NAME)
+
+
+def extract_data_from_csv(config, params):
+    try:
+        endpoint = "files"
+        logger.info(params)
+        file_iri = handle_params(params)
+        file_path = join('/tmp', download_file_from_cyops(file_iri)['cyops_file_path'])
+        logger.info("file from code:" + file_path)
+        if params.get('columnNames'):
+            columnNames = params.get('columnNames')
+            columNames = columNames.split(",")
+            df = pd.read_csv('{}'.format(file_path), delimiter=',', encoding="utf-8-sig")[[columNames]]
+        else:
+            df = pd.read_csv('{}'.format(file_path), delimiter=',', encoding="utf-8-sig")
+
+        if params.get('deDupValuesOn'):
+            deDupValuesOn = params.get('deDupValuesOn')
+            deDupValuesOn = deDupValuesOn.split(",")
+            df.drop_duplicates(subset=[deDupValuesOn], keep='first')
+
+        smaller_datasets = np.array_split(df, 20)
```

Comment:
why split when we need to join again?

Reply:
The number of records returning is too large at times when CSV file is large one , hence to prevent playbook for crashing , we can let user loop returned records over a number of smaller data set. But we now have bulk insert feed , may be I will be take it as an option in operation . Please advice @cs-pooja-singh